runtime/tchannel_server.go

```diff
@@ -72,8 +68,7 @@ func NewTChannelServer(opts *TChannelServerOptions) (*TChannelServer, error) {
 			err)
 	}
 
-	return &TChannelServer{
-		Channel: channel,
-		Logger:  opts.Logger,
-	}, nil
+	gateway.Channel = channel
```

Comment:
mutating the gateway here is naughty, can we move this into `setupChannel()`.

Keeping mutation of the gateway to gateway.go makes debugging easier.